src/utils/validation.spec.ts

```diff
@@ -126,13 +126,18 @@ describe('validation/validation', () => {
     assert.strictEqual(result, true);
   });
 
-  it('isValidGuid returns true with @meid token and spaces', () => {
-    const result = validation.isValidGuid('@meid ');
+  it('isValidUserPrincipalName returns true if valid username', () => {
+    const result = validation.isValidUserPrincipalName('John@Contoso.com');
     assert.strictEqual(result, true);
   });
 
-  it('isValidGuid returns true with @meId (case sensitive)', () => {
-    const result = validation.isValidGuid('@meId ');
+  it('isValidUserPrincipalName returns false if invalid username', () => {
+    const result = validation.isValidUserPrincipalName('foo');
+    assert(result === false);
+  });
+
+  it('isValidUserPrincipalName returns true with @meusername token', () => {
+    const result = validation.isValidUserPrincipalName('@meusername');
     assert.strictEqual(result, true);
   });
```

Comment:
In fact we need 3 tests for each @me token.

- One with the @me token
- One with the @me token and cases
- One with the @me token and spaces